scripts/keeper.ts

```diff
@@ -1,156 +1,127 @@
-import { Event, Signer } from "ethers";
+import { BigNumber, Signer } from "ethers";
 import { isEqual } from "lodash";
 import {
   L1DAITeleportGateway,
   Starknet,
-  TeleportJoin,
 } from "types/ethers-contracts";
+import type {
+  ConsumedMessageToL1Event,
+  LogMessageToL1Event,
+} from "types/ethers-contracts/Starknet";
 import { l2_dai_teleport_gateway } from "types/starknet-contracts";
 
 import {
   cairoShortStringToBytes32,
   Config,
-  findNearestBlock,
   getConfig,
   getL1ContractAt,
   getL1Signer,
   getL2ContractAt,
   getL2Signer,
-  l1String,
-  l2String,
+  getL1TeleportGatewayAddress,
   toUint,
+  findNearestBlock,
 } from "./utils";
 
-export async function flush(config: Config) {
-  const l1Signer = getL1Signer(config);
-  const l2Signer = getL2Signer(config);
+const HANDLE_FLUSH = BigNumber.from(1);
+const FINALIZE_FLUSH = BigNumber.from(1);
 
-  const l2TeleportGateway = await getL2ContractAt<l2_dai_teleport_gateway>(
-    l2Signer,
-    "l2_dai_teleport_gateway",
-    config.l2TeleportGatewayAddress
+async function getDomainsToFlush(
+  l2TeleportGateway: l2_dai_teleport_gateway,
+  l1Signer: Signer,
+  { flushDelay, flushDelayMultiplier, starknetAddress }: Config
+): Promise<string[]> {
+  const l1TeleportGatewayAddress = `0x${(await l2TeleportGateway.teleport_gateway())[0].toString(16)}`;
+  const starknet = await getL1ContractAt<Starknet>(
+    l1Signer,
+    "Starknet",
+    starknetAddress
   );
-
-  const encodedTargetDomain = l2String(config.targetDomain);
-
-  const lastFlushTimestamp = await recentFlushTimestamp(config, l1Signer);
-  const [daiToFlushSplit] = await l2TeleportGateway.batched_dai_to_flush(
-    encodedTargetDomain
+  const logMessageFilter = starknet.filters.LogMessageToL1(
+    l2TeleportGateway.address,
+    l1TeleportGatewayAddress
   );
-  const daiToFlush = toUint(daiToFlushSplit);
-  console.log(`DAI to flush: ${daiToFlush}`);
+  const logMessageEvents = await starknet.queryFilter(
+    logMessageFilter,
+    findNearestBlock(l1Signer.provider, Date.now() - flushDelayMultiplier * flushDelay)
+  );
+  const flushEvents = logMessageEvents.filter(event => {
+    return event.args[0] === FINALIZE_FLUSH;
+  });
 
-  if (
-    daiToFlush > config.flushMinimum &&
-    Date.now() > lastFlushTimestamp + config.flushDelay
-  ) {
-    console.log("Sending `flush` transaction");
-    const { amount } = await l2TeleportGateway.estimateFee.flush([
-      encodedTargetDomain
-    ]);
-    const { transaction_hash } = await l2TeleportGateway.flush(
-      encodedTargetDomain,
-      { maxFee: amount * config.l2GasMultiplier }
-    );
-    await l2Signer.waitForTransaction(transaction_hash);
-    console.log("Success");
-  } else {
-    console.log("Flush not needed");
-  }
+  // eventually get targetDomains from File events
+  const targetDomains = ["GOERLI-MASTER-1"];
+  const domainsToFlush = [];
```

Comment:
Shouldn't `targetDomains` be used to initialize domainsToFlush? I mean to sth like:
```
{ 
  'GOERLI-MASTER-1': 0 // 0 is previous flush timestamp, should be updated by flushEvents query 
}
```
?